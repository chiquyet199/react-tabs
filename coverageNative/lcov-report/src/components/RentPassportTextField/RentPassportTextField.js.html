<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/RentPassportTextField/RentPassportTextField.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/RentPassportTextField</a> RentPassportTextField.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">63.64% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>14/22</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">46.43% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>13/28</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">30% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.9% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>13/21</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">17x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from "react";
import styled from "styled-components/native";
import {colors} from "../../constants/theme";
import {bool} from "prop-types";
&nbsp;
export const FlexContainer = styled.View`
  flex: auto;
  flex-direction: row;
  flex-wrap: wrap;
  width: ${<span class="fstat-no" title="function not covered" >pr</span>ops =&gt; <span class="cstat-no" title="statement not covered" >props.width}</span>%;
  padding-vertical: 10px;
`;
&nbsp;
export const PageHeader = styled.Text`
  font-size: ${<span class="fstat-no" title="function not covered" >({</span>fontSize}) =&gt; <span class="cstat-no" title="statement not covered" >fontSize || "18"}</span>px;
  display: flex;
  font-style: ${<span class="fstat-no" title="function not covered" >({</span>fontStyle}) =&gt; <span class="cstat-no" title="statement not covered" >fontStyle || "normal"}</span>;
  width: ${<span class="fstat-no" title="function not covered" >({</span>align}) =&gt; (<span class="cstat-no" title="statement not covered" >align === "right" ? "auto" : "100%")</span>};
  color: ${<span class="fstat-no" title="function not covered" >({</span>error}) =&gt;
<span class="cstat-no" title="statement not covered" >    error === "true" ? colors.alertRed : colors.canopySteel}</span>;
  align-self: flex-end;
  align-items: flex-end;
  justify-content: ${<span class="fstat-no" title="function not covered" >({</span>align}) =&gt;
<span class="cstat-no" title="statement not covered" >    align === "right" ? "space-between" : "flex-start"}</span>;
`;
&nbsp;
export const TextFieldHeading = styled.Text`
  font-size: 14px;
  font-weight: bold;
  color: ${<span class="fstat-no" title="function not covered" >({</span>error}) =&gt;
<span class="cstat-no" title="statement not covered" >    error === "true" ? colors.alertRed : colors.canopySteel}</span>;
  width: 100%;
  display: flex;
`;
&nbsp;
class RentPassportTextField extends React.Component {
  joinN = value =&gt; {
    if (Array.isArray(value) &amp;&amp; value) {
      return value.filter(n =&gt; n).join("\n");
    } else if (typeof value === "string" &amp;&amp; value.trim().length) {
      return value;
    } else <span class="missing-if-branch" title="if path not taken" >I</span>if (typeof value === "number" || typeof value === "object") {
<span class="cstat-no" title="statement not covered" >      return value;</span>
    }
&nbsp;
    return "";
  };
&nbsp;
  render() {
    const {title, value, width, fontStyle, fontSize, align, error} = this.props;
    const errorProp = error ? error.toString() : <span class="branch-1 cbranch-no" title="branch not covered" >undefined;</span>
&nbsp;
    return (
      &lt;FlexContainer width={width} align={align}&gt;
        {title &amp;&amp; (
<span class="branch-1 cbranch-no" title="branch not covered" >          &lt;TextFieldHeading error={errorProp} align={align}&gt;</span>
            {title}
          &lt;/TextFieldHeading&gt;
        )}
        &lt;PageHeader
          fontStyle={fontStyle}
          fontSize={fontSize}
          align={align}
          error={errorProp}
          textAlignVertical="bottom"
        &gt;
          {this.joinN(value)}
        &lt;/PageHeader&gt;
      &lt;/FlexContainer&gt;
    );
  }
}
&nbsp;
RentPassportTextField.propTypes = {
  error: bool,
};
&nbsp;
export default RentPassportTextField;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 28 2019 14:23:50 GMT+0700 (Indochina Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
