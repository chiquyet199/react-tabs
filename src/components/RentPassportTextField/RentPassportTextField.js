import React from "react";
import styled from "styled-components/native";
import {colors} from "../../constants/theme";
import {bool} from "prop-types";

export const FlexContainer = styled.View`
  flex: auto;
  flex-direction: row;
  flex-wrap: wrap;
  width: ${props => props.width}%;
  padding-vertical: 10px;
`;

export const PageHeader = styled.Text`
  font-size: ${({fontSize}) => fontSize || "18"}px;
  display: flex;
  font-style: ${({fontStyle}) => fontStyle || "normal"};
  width: ${({align}) => (align === "right" ? "auto" : "100%")};
  color: ${({error}) =>
    error === "true" ? colors.alertRed : colors.canopySteel};
  align-self: flex-end;
  align-items: flex-end;
  justify-content: ${({align}) =>
    align === "right" ? "space-between" : "flex-start"};
`;

export const TextFieldHeading = styled.Text`
  font-size: 14px;
  font-weight: bold;
  color: ${({error}) =>
    error === "true" ? colors.alertRed : colors.canopySteel};
  width: 100%;
  display: flex;
`;

class RentPassportTextField extends React.Component {
  joinN = value => {
    if (Array.isArray(value) && value) {
      return value.filter(n => n).join("\n");
    } else if (typeof value === "string" && value.trim().length) {
      return value;
    } else if (typeof value === "number" || typeof value === "object") {
      return value;
    }

    return "";
  };

  render() {
    const {title, value, width, fontStyle, fontSize, align, error} = this.props;
    const errorProp = error ? error.toString() : undefined;

    return (
      <FlexContainer width={width} align={align}>
        {title && (
          <TextFieldHeading error={errorProp} align={align}>
            {title}
          </TextFieldHeading>
        )}
        <PageHeader
          fontStyle={fontStyle}
          fontSize={fontSize}
          align={align}
          error={errorProp}
          textAlignVertical="bottom"
        >
          {this.joinN(value)}
        </PageHeader>
      </FlexContainer>
    );
  }
}

RentPassportTextField.propTypes = {
  error: bool,
};

export default RentPassportTextField;
