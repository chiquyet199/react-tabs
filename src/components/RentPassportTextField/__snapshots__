/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/RentPassportTextField/__native_snapshots__