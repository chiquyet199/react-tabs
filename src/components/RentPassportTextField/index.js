import RentPassportTextField from "./RentPassportTextField";

export default RentPassportTextField;
