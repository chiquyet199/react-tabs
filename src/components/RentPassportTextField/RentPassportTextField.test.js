import {Platform} from "react-native";
import React from "react";
import {shallow} from "enzyme";

import configureStore from "redux-mock-store";
import Translations from "../../utils/locales/en-US";
import RentPassportTextField, {PageHeader} from "./RentPassportTextField";

const mockProps = {
  title: Translations.what_you_do,
  value: ["Employed", "Employed", "Employed"],
  width: 50,
  fontStyle: "none",
  fontSize: 18,
  align: "right",
  error: true,
};
const value = "n/a";
const testOutputString = mockProps.value.join("\n");
const mockStore = configureStore(mockProps);
const textFieldComp = <RentPassportTextField {...mockProps} />;
const wrapper = shallow(textFieldComp, {
  context: {store: mockStore(mockProps)},
});

const mockPropsValues = [
  ["Employed", "Employed", "Employed"],
  ["Employed", "Employed", "Employed", "Employed", "Employed"],
  "          ",
  "",
];

const expectedValues = [
  mockPropsValues[0].join("\n"),
  mockPropsValues[1].join("\n"),
  "",
  "",
];

describe("RentPassportTextField component", () => {
  it("renders properly - " + Platform.OS, () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("check if called function joinN", () => {
    const spy = jest.spyOn(wrapper.instance(), "joinN");
    wrapper.instance().forceUpdate();
    expect(spy).toHaveBeenCalledWith(mockProps.value);
  });

  it("check RentPassportTextField with default value", () => {
    const wrapperNA = shallow(<RentPassportTextField {...mockProps} />);
    expect(wrapperNA.find(PageHeader).props().children).toEqual(
      testOutputString,
    );
  });

  it("check RentPassportTextField with n/a value", () => {
    mockProps.value = value;
    const wrapperNA = shallow(<RentPassportTextField {...mockProps} />);
    expect(wrapperNA.find(PageHeader).props().children).toEqual(value);
  });

  it("Check if in PageHeader has a array", () => {
    mockPropsValues.forEach((item, index) => {
      mockProps.value = item;
      const wrapperNA = shallow(<RentPassportTextField {...mockProps} />);
      expect(wrapperNA.find(PageHeader).props().children).toEqual(
        expectedValues[index],
      );
    });
  });
});
