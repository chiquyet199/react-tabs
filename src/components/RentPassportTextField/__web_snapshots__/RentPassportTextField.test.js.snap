// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RentPassportTextField component renders properly - web 1`] = `
<Styled(View)
  align="right"
  width={50}
>
  <Styled(Text)
    align="right"
    error="true"
    fontSize={18}
    fontStyle="none"
    textAlignVertical="bottom"
  >
    Employed
Employed
Employed
  </Styled(Text)>
</Styled(View)>
`;
